apiclient/rancher_api/api.py

```diff
@@ -56,8 +56,16 @@ def cluster_version(self):
         if not self._version:
             code, data = self.settings.get('server-version')
             ver = data['value']
-            # XXX: fix master-xxx-head to 8.8.8, need the API fix the problem
-            self._version = parse_version('8.8.8' if 'master' in ver else ver)
+
+            try:
+                # XXX: fix master-xxx-head to 8.8.8, need the API fix the problem
+                cur, _, _ = ver.split('-')
+                cur = "v8.8.8" if "master" in cur else cur
+                self._version = parse_version(f"{cur}.99")
+            except ValueError:
+                self._version = parse_version(ver)
```

Comment:
It's better to keep current implementation in try and new one in exception.
```
try:
    self._version = parse_version('8.8.8' if 'master' in ver else ver)
except:
    #YOUR_NEW_IMPLEMENTATION
```

Reply:
No, why?